api/apps/api/src/modules/clone/export/domain/export/export.ts

```diff
@@ -0,0 +1,107 @@
+import { v4 } from 'uuid';
+import { AggregateRoot } from '@nestjs/cqrs';
+import { Either, left, right } from 'fp-ts/Either';
+
+import { ResourceKind } from './resource.kind';
+import { ExportId } from './export.id';
+import { ResourceId } from './resource.id';
+import { ArchiveLocation } from './archive-location';
+
+import { ClonePartRequested } from '../events/clone-part-requested.event';
+import { ClonePartsFinished } from '../events/clone-parts-finished.event';
+import { ArchiveReady } from '../events/archive-ready.event';
+
+import { PieceLocation } from './clone-part/piece-location';
+import { ClonePart } from './clone-part/clone-part';
+import { PieceId } from './clone-part/piece.id';
+import { ExportSnapshot } from './export.snapshot';
+
+export const pieceNotFound = Symbol('export piece not found');
+export const notReady = Symbol('some pieces of export are not yet ready');
+
+export class Export extends AggregateRoot {
+  private constructor(
+    public readonly id: ExportId,
+    public readonly resourceId: ResourceId,
+    public readonly resourceKind: ResourceKind,
```

Comment:
Are you sure that kind is a good discriminator? Where it will affect the app logic?

Reply:
What you would propose?